pnet_sys/src/unix.rs

```diff
@@ -107,10 +107,10 @@ pub mod public {
     }
 
     fn make_in6_addr(segments: [u16; 8]) -> In6Addr {
-        #[allow(deprecated)]
-        let mut val: In6Addr = unsafe { mem::uninitialized() };
-        val.s6_addr = unsafe {
-            mem::transmute([
+        // Safety: We're transmuting an array of ints to an array of ints.
+        // There is no padding involved, and they must be the same size.
```

Comment:
Is this definitely true on all architectures?

This code could actually be written without the `transmute()` - that's probably a better option.

Reply:
Well, the size of `u16` is fixed at 2 bytes, and the min/max values are fixed too. So there's no *room* for padding. (If it was 2 bytes long but needed to be padded to 4 byte boundaries, it would count as a 4 byte long type).

We could take a `bytemuck` dependency and use that to transmute, or just do it manually with shifts in a loop and hope it gets optimised. I've not benchmarked the latter, it's *probably* fine.